moveit_core/robot_model/src/joint_model_group.cpp

```diff
@@ -97,188 +101,239 @@ bool jointPrecedes(const JointModel* a, const JointModel* b)
 
   return false;
 }
+
+// These are used as filter functions
+const auto is_fixed_joint = [](const auto& jm) { return jm->getVariableCount() == 0; };
+const auto is_mimic_joint = [](const auto& jm) { return jm->getMimic() != nullptr; };
+const auto has_geometry = [](const auto& lm) { return !lm->getShapes().empty(); };
+
+// These are transform functions
+const auto get_name = [](const auto& element) { return element->getName(); };
+const auto get_variable_count = [](const auto& jm) { return jm->getVariableCount(); };
+const auto get_variable_bounds_addr = [](const auto& jm) { return &jm->getVariableBounds(); };
+const auto get_first_variable_index = [](const auto& jm) { return jm->getFirstVariableIndex(); };
+const auto get_variable_names_view = [](const auto& jm) { return views::all(jm->getVariableNames()); };
+const auto get_variable_names_size = [](const auto& jm) { return jm->getVariableNames().size(); };
+
+// Helper to convert a tuple to a pair.  This is useful as a transform after
+// zipping two views inorder to initialize a map.
+const auto tuple_to_pair = [](const auto& element) {
+  return std::make_pair(std::get<0>(element), std::get<1>(element));
+};
 }  // namespace
 
 JointModelGroup::JointModelGroup(const std::string& group_name, const srdf::Model::Group& config,
                                  const std::vector<const JointModel*>& unsorted_group_joints,
                                  const RobotModel* parent_model)
-  : parent_model_(parent_model)
-  , name_(group_name)
-  , common_root_(nullptr)
-  , variable_count_(0)
-  , active_variable_count_(0)
-  , is_contiguous_index_list_(true)
-  , is_chain_(false)
-  , is_single_dof_(true)
-  , config_(config)
-{
-  // sort joints in Depth-First order
-  joint_model_vector_ = unsorted_group_joints;
-  std::sort(joint_model_vector_.begin(), joint_model_vector_.end(), OrderJointsByIndex());
-  joint_model_name_vector_.reserve(joint_model_vector_.size());
+  : parent_model_{ parent_model }
+  , name_{ group_name }
+  , common_root_{ nullptr }
+  , variable_count_{ 0 }
+  , active_variable_count_{ 0 }
+  , is_contiguous_index_list_{ true }
+  , is_chain_{ false }
+  , is_single_dof_{ true }
+  , config_{ config }
+{
+  // Disable clang-format to keep nicer formatting of ranges
+  // clang-format off
+  joint_model_vector_ = unsorted_group_joints
+    | ranges::to<std::vector>()
+    | actions::sort(OrderJointsByIndex());
```

Comment:
```suggestion
    | actions::sort(OrderJointsByIndex{});
```
Can we use uniform initialization?

Reply:
:+1: